Proposal for fleet management overview page (#2185)
giantswarm/docs: src/content/overview/fleet management/_index.md

  principal:
    parent: overview
    identifier: overview-fleet-management
last_review_date: 2024-04-18
owner:
  - https://github.com/orgs/giantswarm/teams/sig-product
---

Our experience tells us that managing a large microservices platform is not easy. Ideally, workloads are created ready to run in the cloud and are optimized for the container lifecycle. But even so, when you need to manage several providers, regions and environments, the complexity multiplies exponentially. Having that in mind, we have created a some abstractions to help you dealing with that complexity in a more efficient way.

## Features

> The fleet management features that our product offers are:

Reviewer: I'm still on the crusade to remove the word "our" 😉 

```suggestion
The Giant Swarm platform offers these fleet management features:
```

Author: Sorry I will review open PRs to avoid it

> - **Multi-environment**: We support multiple providers, regions and environments. Platform engineers can map their company structure to our platform structure in order to ease the management of their teams and workloads.

Reviewer: Changing this to be written from customer perspective ("you"). I don't understand the second part (map what?).

```suggestion
- **Multi-environment**: Deploy to multiple providers, regions and environments. Platform engineers can [[[map their company structure to our platform structure]]] <== I HAVE NO IDEA WHAT THIS MEANS in order to ease the management of their teams and workloads.
```

> - **Configuration management**: We rely on GitOps principles to manage the configuration of your clusters, environments and workloads having multiple layers of configuration.

Reviewer: ```suggestion
- **Configuration management**: Manage configuration of your clusters, environments and workloads with GitOps, a versioned, single source of truth repository that shows the currently deployed state.
```

I've removed the "multiple layers" part since it can sound like complexity. Maybe you have an idea to instead explain the benefit in one sentence? Such as: making environments consistent, yet flexibly updateable (I didn't find a good sentence here).

> - **Cluster lifecycle management**: We provide a way to manage the lifecycle of your clusters, from creation to deletion, including upgrades and scaling.

Reviewer: ```suggestion
- **Cluster lifecycle management**: Creation and deletion of clusters is only a matter of minutes. Giant Swarm offers well-tested releases so make upgrades smooth and fully-automated. Scaling features for clusters are built-in.
```

> - **Workload management**: We have developed a solution to help your developers to configure and deploy their applications in a cloud-native way.

Reviewer: Too vague. What do we describe here? Is it the app platform? If yes, does "workload management" really cut it as a good term?

Author: yeah I asked Puja for help to define those better, I came up with some ideas but ideally we describe those better :)

## Cloud-native applications

> We have rely on Cluster API implementation to manage the lifecycle of your clusters. Together with the app platform, we provide a way to manage your workloads in a cloud-native way.

Reviewer: I'm not sure yet what the purpose of this section in the template should be? Do we want a list of technologies here, so deciders can see the complexity and choices (then probably a heading "Technology choices", "Cloud-native technology" or so might fit a little better)? Or do we have other goals here? With that, I could review these better.

```suggestion
The Giant Swarm platform uses [Cluster API](https://cluster-api.sigs.k8s.io/), an open-source, well-supported subproject of Kubernetes, to manage the lifecycle of your clusters. Together with the app platform, we provide a way to manage your workloads in a [[[cloud-native way]]] <= VAGUE TERM WHICH NEWCOMERS TO THE K8S WORLD MAY NOT UNDERSTAND LET'S RATHER EXPLAIN BENEFITS EXPLICITLY.
```
